images/egress/dns-proxy/Dockerfile

```diff
@@ -0,0 +1,28 @@
+#
+# This is the egress router L4 DNS proxy for OpenShift Origin
+#
+# The standard name for this image is openshift/origin-egress-dns-proxy
+
+FROM openshift/origin-base
+
+# HAProxy 1.6+ version is needed to leverage DNS resolution at runtime.
+RUN INSTALL_PKGS="rsyslog gcc make openssl-devel pcre-devel tar wget socat" && \
+    yum install -y $INSTALL_PKGS && \
+    rpm -V $INSTALL_PKGS && \
+    wget http://www.haproxy.org/download/1.7/src/haproxy-1.7.5.tar.gz && \
+    tar xvzf haproxy-1.7.5.tar.gz && \
+    groupadd haproxy && \
+    useradd -g haproxy haproxy && \
+    cd haproxy-* && make TARGET=linux2628 CPU=native USE_PCRE=1 USE_OPENSSL=1 USE_ZLIB=1 && make install && \
```

Comment:
Dynamic DNS resolution feature is only available in HAProxy 1.6+ version (currently we are running haproxy 1.5). I don't know if this method is acceptable currently or not but we have done similar stuff in release-1.3 (https://github.com/openshift/origin/blob/release-1.3/images/router/haproxy/Dockerfile)